### DISCUSSION TITLE
What does it take to maintain an app like this?

### DISCUSSION THREAD

**Discussion by ghxstweb** [Schneegans/Fly-Pie#342] [Fly-Pie Q&A]
Hi simon. I was just curious, now that flypie is a relatively mature app and feature complete, what do you have to do regularly to maintain this app? Is it very demanding or do you simply have to update versions, support dependency imports, etc.? Thanks in advance for the clarification!

**Comment by Schneegans** [comment]
Hi there! Well, that's difficult to answer. Currently, I am not investing much time into Fly-Pie because I have other projects which are higher on my priority list. [Kando](https://github.com/kando-menu/kando) will eventually become the successor to Fly-Pie and [Burn-My-Windows](https://github.com/Schneegans/Burn-My-Windows) has at least a magnitude more users than Fly-Pie. So I _could_ invest much more time - there are some pretty [annoying and long-standing issues](https://github.com/Schneegans/Fly-Pie/issues). But for most users, it's working decently, I suppose.

In general, GNOME Shell extensions require quite a huge amount of maintenance. Every six months a new version of GNOME Shell is released and most of the time extensions needs to be adapted to changes in GNOME Shell's source code. Especially if you want to make your code backwards-compatible to older GNOME Shell versions a lot of testing is required. [CI jobs can help a lot here](https://schneegans.github.io/tutorials/2022/02/28/gnome-shell-extensions-ci-01). Back in the days, I chose to create Fly-Pie as a GNOME Shell extension simply because something like this was impossible to implement in a stand-alone manner on Wayland. Today, I would highly recommend to create something as a GNOME Shell extension only if it is really necessary. A standalone app definitely comes with less maintenance overhead.

Over the last couple of months, I would guess that I invested maybe about 15h / week into my open source projects. Out of these, maybe one hour was for Fly-Pie. The majority goes into Kando these days, but I could spend much more for Fly-Pie if I had the time :sweat_smile:

**Reply by ghxstweb** [reply]
Thanks Simon! When you say standalone app, you mean something in electron, correct? I have yet to check out Kando because flypie is working just fine for me at the moment on GNOME!

How many hours a week do you spend on Kando?

**Reply by Schneegans** [reply]
Kando is written in Electron, but this comes [with some significant disadvantages](https://github.com/orgs/kando-menu/discussions/58). For a Linux-only app, there are definitely better choices. And even for cross-platform apps I would think twice before choosing Electron.

I think that I invest about 10h / week for Kando currently.
